src/main/java/us/kilroyrobotics/subsystems/Tower.java

```diff
@@ -0,0 +1,288 @@
+import us.kilroyrobotics.util.TowerEvent;
+import us.kilroyrobotics.util.TowerState;
+
+public class Tower extends SubsystemBase {
+
+    private TowerState currentState = TowerState.INIT;
+    private TowerEvent pendingEvent = TowerEvent.NONE;
+
+    private final Timer stateTimer = new Timer();
+
+    private final CommandSwerveDrivetrain drivetrain;
+    private final SwerveRequest.RobotCentric forwardStraight;
+    private final SwerveRequest.PointWheelsAt point;
+    private final Elevator elevator;
+    private final Wrist wrist;
+    private final CoralIntakeMotor coralIntakeMotor;
+    private final LEDs leds;
+
+    private final Command backupCommand;
+
+    private double safetyFactor = 1;
+    private int currentLevel = 0;
+
+    /** Creates a new Tower. */
+    public Tower(
+            CommandSwerveDrivetrain drivetrain,
+            SwerveRequest.RobotCentric forwardStraight,
+            SwerveRequest.PointWheelsAt point,
+            Elevator elevator,
+            Wrist wrist,
+            CoralIntakeMotor coralIntakeMotor,
+            LEDs leds) {
+        this.drivetrain = drivetrain;
+        this.forwardStraight = forwardStraight;
+        this.point = point;
+        this.elevator = elevator;
+        this.wrist = wrist;
+        this.coralIntakeMotor = coralIntakeMotor;
+        this.leds = leds;
+
+        backupCommand =
+                Commands.runEnd(
+                                () ->
+                                        drivetrain.setControl(
+                                                forwardStraight.withVelocityX(
+                                                        FeetPerSecond.of(-1))),
+                                () -> drivetrain.setControl(forwardStraight.withVelocityX(0)))
+                        .withTimeout(0.25);
+
+        stateTimer.start();
+    }
+
+    public void initialize() {
+        coralIntakeMotor.setSpeed(0);
+
+        elevator.stop();
+        elevator.resetClosedLoopControl();
+
+        wrist.stop();
+        wrist.resetClosedLoopControl();
+
+        pendingEvent = TowerEvent.NONE;
+        setState(TowerState.INIT);
+    }
+
+    public void runStateMachine() {
+        if (isTriggered(TowerEvent.HOME_TOWER)) setState(TowerState.INIT);
+        if (isTriggered(TowerEvent.SCORE_BYPASS)) {
+            coralIntakeMotor.setSpeed(CoralMechanismConstants.kWheelSpeedOuttaking);
+
+            setState(TowerState.SCORING);
+        }
+        switch (currentState) {
+            case INIT:
+                wrist.setAngle(CoralMechanismConstants.kIntakingAngle);
+                coralIntakeMotor.setSpeed(0.0);
+
+                setState(TowerState.HOMING_WRIST);
+                break;
+            case HOMING_WRIST:
+                if (wrist.inPosition()) setState(TowerState.HOMING_ELEVATOR);
+                break;
+            case HOMING_ELEVATOR:
+                if (elevator.inPosition()) {
+                    elevator.setPosition(ElevatorConstants.kZeroed);
```

Comment:
this needs to be in HOMING_WRIST

Reply:
Wait i just realized why are you waiting for elevator to be in position and then zeroing it?

Reply:
if it ain't broke don't fix it 💯 /hj idk